### PR TITLE
add optional support for Bathroom Epiphanies Frosty Flake rev 20130602

### PR DESCRIPTION
**Pull request by dyindude**
The original code for this board was ported using the TMK code from https://github.com/BathroomEpiphanies/epiphanies_tmk_keyboard/blob/master/be_controllers/config_frosty_flake_20140521.h

There are two revisions for this controller. In this PR I have implemented an option to build for revision 20130602 by referencing the values in https://github.com/BathroomEpiphanies/epiphanies_tmk_keyboard/blob/master/be_controllers/config_frosty_flake_20130602.h

The preprocessor statement is commented out in config.h, so the original build for 20140521 will be default.

I have tested to make sure revision 20140521 compiles. When I flash my Frosty Flake rev 20130602 with this firmware, the keyboard does not operate (same behavior as when building from current master branch). I do not have a revision 20140521 board to test with.

I have tested the revision 20130602 on my own Frosty Flake and verified that my keymap functions as expected.

## Description

Added option `BE_CONTROLLER_REVISION` that is commented out by default in `config.h`

When it is set to `20130602` the changes in `frosty_flake.c` will ensure the correct LED pins are used, and the changes in `matrix.c` will allow the firmware to properly address the matrix on the 20130602 revision controller.

If `BE_CONTROLLER_REVISION` is set to any other value, the firmware will be built for revision 20140521

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — COMMENTED** [review]
This would be better as a revision/subfolder, I think.

**Comment by dyindude** [comment]
I found
https://beta.docs.qmk.fm/developing-qmk/c-development/hardware_keyboard_guidelines#sub-folders
and will use this as an example for setting up the revisions as folders instead

**Comment by dyindude** [comment]
I've reimplemented this using subfolders for the revisions. Please let me know if something else needs to be changed, first time contributing :)

File: keyboards/bpiphany/frosty_flake/20130602/rules.mk [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#
+BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (noroadsleft) [inline keyboards/bpiphany/frosty_flake/20130602/rules.mk:18]:
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

File: keyboards/bpiphany/frosty_flake/20140521/rules.mk [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+#
+BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (noroadsleft) [inline keyboards/bpiphany/frosty_flake/20140521/rules.mk:18]:
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
Just some minor clean-up of the `rules.mk` files. Looks good to me otherwise.

File: keyboards/bpiphany/frosty_flake/20140521/rules.mk (hunk 2 above)
Comment (dyindude) [reply keyboards/bpiphany/frosty_flake/20140521/rules.mk]: These changes have been implemented in the next commit. Thanks!

**Comment by noroadsleft** [comment]
Merged our `master` branch into your submitted branch to fix a merge conflict. We're good to go; thanks!

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Comment by tofurky** [comment]
@dyindude this swaps the num lock and caps lock LEDs for the 2014 variant. https://github.com/qmk/qmk_firmware/compare/master...tofurky:frosty_flake_capslock_led may fix it but i cannot test the 2013 variant.

**Comment by tofurky** [comment]
@dyindude i opened https://github.com/qmk/qmk_firmware/pull/16000 to hopefully fix the mixup. can you please test the 2013 variant?

**Comment by dyindude** [comment]
@tofurky My 2013 does not have any LEDs (I removed them/broke the leds when swapping switches), but I was able to test to make sure that there was no functional regressions from this change

**Pull request merged**